axlearn/common/learner.py

```diff
@@ -444,6 +444,153 @@ def _mask_tree(tree: dict, *, keep: dict) -> dict:
     )
 
 
+class MetricsAccumulationOp(NamedTuple):
```

Comment:
Axlearn already has metric accumulation classes that are used by evalers. Could those be reused here instead of defining new classes?

Reply:
These metric accumulation classes are stateless to be usable as carry by jax.lax.scan unlike the ones in the evaler, I can make the class structure similar though